### PR TITLE
Feature/channellist

### PR DESCRIPTION
**Pull request by adrian09h**
# Submit a pull request

## Fix channels unread mark
- remove `sortUserReads()` from `ChannelState`
- add `sort` to `ChannelStateGsonAdapter ` instead.
```java
public class ChannelStateGsonAdapter extends TypeAdapter<ChannelState> {
   ...
    @Override
    public ChannelState read(JsonReader in) throws IOException {
        ...
        if (value.getReads() != null && !value.getReads().isEmpty()){
            Collections.sort(value.getReads(), (ChannelUserRead o1, ChannelUserRead o2) -> o1.getLastRead().compareTo(o2.getLastRead()));
        }
        return value;
    }
}
```

**Result**

### CODE REVIEW

File: library/src/main/java/com/getstream/sdk/chat/viewmodel/ChannelViewModel.java [hunk 1]
```diff
@@ -715,7 +716,7 @@ public void onSendMessage(final Message message, @Nullable final MessageCallback
                 callback.onError("no internet", -1);
             return;
         }
-
+        channel.getChannelState().setReadDateOfChannelLastMessage(client().getUser(), message.getCreatedAt());
```
Comment (tschellenbach) [inline library/src/main/java/com/getstream/sdk/chat/viewmodel/ChannelViewModel.java]: why is this here? it looks like some debug code....

Comment (adrian09h) [reply library/src/main/java/com/getstream/sdk/chat/viewmodel/ChannelViewModel.java]: No debug. After sent message, we must add current user immediately to reads.

Comment (tschellenbach) [reply library/src/main/java/com/getstream/sdk/chat/viewmodel/ChannelViewModel.java]: we do this differently in other sdks, something to be said for this though:

@tbarbugli need your input on how we handle this across SDKs
